#!/usr/bin/env python
PACKAGE = "control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("p_speed",     double_t, 0, "p",      0, 0, 10)
gen.add("i_speed",     double_t, 0, "i",      0, 0, 10)
gen.add("d_speed",     double_t, 0, "d",      0, 0, 10)
gen.add("i_min",       double_t, 0, "i min",  -10, -10, 10)
gen.add("i_max",       double_t, 0, "i max",  -10, -10, 10)
gen.add("use_antiwindup",       bool_t, 0, "use_antiwindup", True)

gen.add("use_brake",       bool_t, 0, "use engine brake for braking", False)

gen.add("braking_threshold",     double_t, 0, "if decelerating with a larger output than braking_threshold scale the pid output with brake_output_scaling if use_brake is true",      2, 0, 20)

gen.add("brake_output_scaling",     double_t, 0, "if decelerating with a larger output than braking_threshold scale the pid output with brake_output_scaling",      2, 0, 20)

exit(gen.generate(PACKAGE, "speed_controller", "SpeedController"))
